ewatercycle/models/wflow.py

```diff
@@ -0,0 +1,130 @@
+            given, will be copied to work_dir. If not given, it should already
+            be there.
+
+        Returns:
+            Path to config file and path to config directory
+        """
+        # work_dir = ....
+        config_file = work_dir / inifile.name
+
+        # For Wflow we have to copy the input (which is unwritable)
+        # to a working directory (what is writable)
+        shutil.copytree(src=parameterset, dst=work_dir)
+        shutil.copy(src=forcing_file, dst=work_dir)
+
+        # Modify the path to the forcing data in the config file
+        cfg = ConfigParser()
+        cfg.optionxform = lambda x: x
+        cfg.read(inifile)
+        cfg.set("framework", "netcdfinput", forcing_file.name)
+        with open(config_file, "w") as filename:
+            cfg.write(filename)
+
+        # Start the container
+        if CFG.container_engine.lower() == "docker":
+            self.bmi = BmiClientDocker(
+                image=CFG["docker_images.wflow"],
+                image_port=55555,
+                work_dir=work_dir,
+            )
+        elif CFG.container_engine.lower() == "singularity":
+            self.bmi = BmiClientSingularity(
+                image=CFG["singularity_images.wflow"],
+                work_dir=work_dir,
+            )
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+
+        return config_file
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        # TODO this is just copied from the comparison notebook.
+        # TODO update this so it can work as standalone code
+
+        # Let's analyse runoff
+        variable = "RiverRunoff"
+
+        # Get time information
+        time_units = model.get_time_units()
+        now = num2date(model.get_current_time(), time_units)
+
+        # Get data (and convert to something that's nice in xarray)
+        grid = model.get_var_grid(variable)
+
+        # Get the model's internal configuration
+        lat = model.get_grid_x(grid)
+        lon = model.get_grid_y(grid)
+        shape = model.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        data = np.reshape(model.get_value(variable), shape)
+        da = xr.DataArray(
+            data,
+            coords={
+                "longitude": lon,
+                "latitude": lat,
+                "time": now
+            },
+            dims=["latitude", "longitude"],
+            name=variable,
+            attrs={"units": model.get_var_units(variable)},
+        )
+
+        da.data = np.reshape(model.get_value(variable), shape)
+        da.time.values = num2date(model.get_current_time(), time_units)
+        return da.where(da != -999)
+
+    @property
+    def parameters(self):
+        """List the variable names that are available for this model."""
+        return self.model.get_output_var_names()
```

Comment:
This property was modeled after pymt, there it is used to default values for the setup() method. So the usage here is not inline with pymt.

I think it should return something like
```python
{
 'RiverRunoff': '2, m^3/s'm',
}
```
Was thinking about 
* start/end/step time, but those are restricted by what is available in parameterset/forcing, so having a default does not make sense and you already have getters for them.
* paths (to files in parameterset, forcing_data) in the config file that need to reflect current session, but those are dont have defaults. However pymt docs says [*Check the parameters property to verify that the model inputs
have been updated.*](https://github.com/csdms/pymt/blame/master/docs/usage.rst#L153-L154), could make sense to set them after setup() method so they can be retrieved later.
* reinit flag or warmup period

Reply:
Oh right, that reference is very useful, I hadn't looked at that yet!

Reply:
So this can also be used to interactively set parameters during the model setup phase... However, for wflow the parameters (like RiverRunoff) are not simple values but entire spatial fields, more like initial conditions... Not sure how this translates to the PyMT philosophy.

Reply:
I now made it such that it shows the config parameters that are set through the ini file. This only works after the `model.setup()` is run, but I don't think we can get any closer to PyMT given the constraints of our models.